test/Makefile.autosetup

```diff
@@ -648,6 +648,7 @@ TEST_OBJS	= test/main.o test/common.o \
 		  $(TAGS_OBJS) \
 		  $(THREAD_OBJS) \
 		  $(URL_OBJS)
+ALLOBJS+=   $(TEST_OBJS)
```

Comment:
This line seems unnecessary.
It forces building the test code.

Removing it, both builds behave correctly:
```sh
./configure --compile-commands
./configure --compile-commands --testing
```

Reply:
Not anymore :)